lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -185,6 +187,149 @@ def default_filter_list_out
       config_get_default('ospf_area', 'filter_list_out')
     end
 
+    # CLI can be the following or none
+    # area 1.1.1.1 nssa
+    # the above command can be appended with no-summary and/or
+    # no-redistribution and/or default-information-originate.
+    # route-map <map> can be appended with default-information-originate
+    # Basically, every property this CLI configures is optional
+    def nssa
+      hash = {}
+      output = config_get('ospf_area', 'nssa', @get_args)
+      return hash if output.nil?
+      output.each do |line|
+        next if line.include?('translate')
+        hash[:nssa_enable] = true
+        hash[:no_summary] = true if line.include?('no-summary')
+        hash[:no_redistribution] = true if line.include?('no-redistribution')
+        hash[:def_info_origin] = true if
+          line.include?('default-information-originate')
+        if line.include?('route-map')
+          params = line.split
+          hash[:route_map] = params[params.index('route-map') + 1]
+        end
+      end
+      hash
+    end
+
+    def nssa_enable
+      hash = nssa
+      ne = hash[:nssa_enable] ? hash[:nssa_enable] : default_nssa_enable
```

Comment:
I would keep this simple and just do:

```
    def nssa_enable
      nssa[:nssa_enable].nil? ? false : true
    end
```

...but I still want to know why it's needed at all.